src/Microsoft.ComponentDetection.Detectors/rust/RustCliDetector.cs

```diff
@@ -187,9 +203,81 @@ private static bool ShouldFallbackFromError(string error)
         return true;
     }
 
-    private static bool ParseDependency(string dependency, out string packageName, out string version, out string source)
+    private static bool ParseDependencyMetadata(string dependency, out string packageName, out string version, out string source)
     {
-        var match = DependencyFormatRegex.Match(dependency);
+        // There are a few different formats for pkgids: https://doc.rust-lang.org/cargo/commands/cargo-pkgid.html#description
```

Comment:
There must be a better way to do this, was pressed for time but if there are any suggestions here I'm all ears